pkg/controllers/management/drivers/kontainerdriver/kontainerdriver.go

```diff
@@ -132,6 +133,57 @@ func (l *Lifecycle) download(obj *v3.KontainerDriver) (*v3.KontainerDriver, erro
 }
 
 func (l *Lifecycle) createDynamicSchema(obj *v3.KontainerDriver) error {
+	resourceFields, err := l.getResourceFields(obj)
+	if err != nil {
+		return err
+	}
+
+	dynamicSchema := &v3.DynamicSchema{
+		Spec: v3.DynamicSchemaSpec{
+			SchemaName:     getDynamicTypeName(obj),
+			ResourceFields: resourceFields,
+		},
+	}
+	dynamicSchema.Name = strings.ToLower(getDynamicTypeName(obj))
+	dynamicSchema.OwnerReferences = []v13.OwnerReference{
+		{
+			UID:        obj.UID,
+			Kind:       obj.Kind,
+			APIVersion: obj.APIVersion,
+			Name:       obj.Name,
+		},
+	}
+	dynamicSchema.Labels = map[string]string{}
+	dynamicSchema.Labels[obj.Name] = obj.Status.DisplayName
+	dynamicSchema.Labels[driverNameLabel] = obj.Status.DisplayName
+	_, err = l.dynamicSchemas.Create(dynamicSchema)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return errorsutil.WithMessage(err, "error creating dynamic schema")
+	}
+
+	return l.createOrUpdateKontainerDriverTypes(obj)
+}
+
+func (l *Lifecycle) updateDynamicSchema(obj *v3.KontainerDriver) error {
+	dynamicSchemas, err := l.dynamicSchemasLister.Get("", strings.ToLower(getDynamicTypeName(obj)))
```

Comment:
two things:
1. why is this variable pluralized?
2. when you get something via a lister, you are getting it from the cache and if you intend to modify it, you _MUST_ copy it via deepCopy. Take a few to think/look through other controllers youve written. If you're doing it elsewhere, it should be fixed

Reply:
1. Oversight, I corrected it.
2. Done.  I also found 2 more issues with this in the kontainerdriver.go controller and I've included them with the PR.